File: backend/app/contracts/evidence.py
from typing import Optional, List, Dict, Any
from proofhire.backend.app.schemas import CoreModel, TimestampModel


class EvidenceBase(CoreModel):
    candidate_id: int
    type: str
    external_id: Optional[str] = None
    content: Dict[str, Any]
    verified: bool = False


class EvidenceCreate(EvidenceBase):
    pass


class EvidenceUpdate(CoreModel):
    verified: Optional[bool] = None
    content: Optional[Dict[str, Any]] = None


class Evidence(EvidenceBase, TimestampModel):
    id: int

File: frontend/src/types/evidence.js
export const evidenceShape = {
  id: "number",
  candidate_id: "number",
  type: "string",
  external_id: "string",
  content: "object",
  verified: "boolean",
};

export const normalizedEvidenceShape = {
  source: "string",
  type: "string",
  content: "object",
  raw_id: "string",
  timestamp: "string",
};
